api/types/usertasks/object.go

```diff
@@ -345,6 +395,61 @@ func validateDiscoverEKSTaskType(ut *usertasksv1.UserTask) error {
 	return nil
 }
 
+func validateDiscoverRDSTaskType(ut *usertasksv1.UserTask) error {
+	if ut.GetSpec().Integration == "" {
+		return trace.BadParameter("integration is required")
+	}
+	if ut.GetSpec().DiscoverRds == nil {
+		return trace.BadParameter("%s requires the discover_rds field", TaskTypeDiscoverRDS)
+	}
+	if ut.GetSpec().DiscoverRds.AccountId == "" {
+		return trace.BadParameter("%s requires the discover_rds.account_id field", TaskTypeDiscoverRDS)
+	}
+	if ut.GetSpec().DiscoverRds.Region == "" {
+		return trace.BadParameter("%s requires the discover_rds.region field", TaskTypeDiscoverRDS)
+	}
+
+	expectedTaskName := TaskNameForDiscoverRDS(TaskNameForDiscoverRDSParts{
+		Integration: ut.Spec.Integration,
+		IssueType:   ut.Spec.IssueType,
+		AccountID:   ut.Spec.DiscoverRds.AccountId,
+		Region:      ut.Spec.DiscoverRds.Region,
+	})
+	if ut.Metadata.GetName() != expectedTaskName {
+		return trace.BadParameter("task name is pre-defined for discover-rds types, expected %s, got %s",
+			expectedTaskName,
+			ut.Metadata.GetName(),
+		)
+	}
+
+	if !slices.Contains(DiscoverRDSIssueTypes, ut.GetSpec().IssueType) {
+		return trace.BadParameter("invalid issue type state, allowed values: %v", DiscoverRDSIssueTypes)
+	}
+
+	if len(ut.Spec.DiscoverRds.Databases) == 0 {
```

Comment:
You use `ut.GetSpec()` before this point, and from here you start using `ut.Spec`. I assume it doesn't matter much as if `ut.Spec` is `nil`, one of the previous validations would have failed due to the zero value, but for consistency I wonder if it might be better to continue to use `ut.GetSpec()` here and a few lines below?